test/controller.jl

```diff
@@ -1,3 +1,50 @@
+@testset "parameterized contacts" begin
+    # Construct a mechanism consisting of a single body which can
+    # rotate about its origin
+    world = RigidBody{Float64}("world")
+    mechanism = Mechanism(world, gravity=SVector(0, 0, -9.81))
+    frame = CartesianFrame3D("body")
+    inertia = SpatialInertia(frame, SDiagonal(1., 1, 1), SVector(0., 0, 0), 10.0)
+    body = RigidBody(inertia)
+    joint = Joint("rx", Revolute(SVector(1., 0, 0)))
+    attach!(mechanism, world, body, joint)
+
+    contactmodel = SoftContactModel(hunt_crossley_hertz(k = 500e3), ViscoelasticCoulombModel(0.8, 20e3, 100.))
+    add_contact_point!(body, Contact.ContactPoint(Point3D(default_frame(body), 0., 0, 0), contactmodel))
+
+    # Make the contact normal a parameter, so that it updates its representation
+    # in body frame to match a fixed orientation in world frame
+    state = MechanismState(mechanism)
+    model = SimpleQP.Model(defaultoptimizer())
+    position = Point3D(default_frame(body), 0., 0, 0)
+    μ = 1.0
+    normal = let state = state, body = body
+        Parameter(model) do
+            transform(state, FreeVector3D(root_frame(state.mechanism), 0., 0, 1), default_frame(body))
+        end
+    end
+    controller_contact = QPControl.ContactPoint{4}(position, normal, μ, state, model)
+    controller_contact.maxnormalforce[] = 1e3
+
+    # Constrain the contact force to be non-zero for testing
+    @constraint(model, controller_contact.force_local.v == [0.0, 0.0, 1.0])
+
+    @objective(model, Minimize, 0 * dot(controller_contact.force_local.v, controller_contact.force_local.v)) # Need to have an objective to avoid throwing "unsupported objective" errors https://github.com/tkoolen/SimpleQP.jl/issues/62
+
+    for θ in linspace(-π, π)
```

Comment:
Could you use `Compat.range` instead?

Reply:
This one got hidden (sorry, that first one shouldn't have been a single comment).